core/domain/user_services.py

```diff
@@ -216,6 +233,76 @@ def validate(self):
                 '%s is not a valid value for the dashboard display '
                 'preferences.' % (self.creator_dashboard_display_pref))
 
+    def populate_from_user_data(self, modifiable_user_data):
+        """Populate the UserSettings domain object using the user data in
+            modifiable_user_data.
+
+        Args:
+            modifiable_user_data: ModifiableUserDataV1. The modifiable user
+                data version_1 object with the data information to be updated.
+
+        Raises:
+            ValidationError. None or empty value is provided for display alias
+                attribute.
+        """
+        if (not modifiable_user_data.display_alias or
+                not isinstance(
+                    modifiable_user_data.display_alias,
+                    python_utils.BASESTRING
+                )
+           ):
+            raise utils.ValidationError(
+                'Expected display_alias to be a string, received %s.' %
+                modifiable_user_data.display_alias
+            )
+        self.display_alias = modifiable_user_data.display_alias
+        self.preferred_language_codes = (
+            modifiable_user_data.preferred_language_codes)
+        self.preferred_site_language_code = (
+            modifiable_user_data.preferred_site_language_code)
+        self.preferred_audio_language_code = (
+            modifiable_user_data.preferred_audio_language_code)
+
+    def convert_to_dict(self):
```

Comment:
Usually we call this to_dict() (see other domain objects in the codebase).

Reply:
Changed now.